### PR TITLE
fix: cleanup eslint errors

### PR DESCRIPTION
**Pull request by RossWilliams**

### CODE REVIEW

File: src/functions/ftp/external-poller/pollers/ftpPoller.ts [hunk 1]
```diff
@@ -26,8 +26,8 @@ export class FtpPoller extends RemotePoller {
     await this.client.access(connectionDetails.config);
   }
 
-  async disconnect(): Promise<void> {
-    this.client.close();
+  disconnect(): Promise<void> {
+    return Promise.resolve(this.client.close());
   }
```
Comment (rsioss) [inline src/functions/ftp/external-poller/pollers/ftpPoller.ts:31]: this change is interesting... perhaps it wouldn't have made this change if the original code was `return this.client.close()`?

File: src/lib/types/RemoteConnectionConfig.ts [hunk 2]
```diff
@@ -45,6 +45,8 @@ export const RemotePollerConfigSchema = z.strictObject({
 
 export type RemotePollerConfig = z.infer<typeof RemotePollerConfigSchema>;
 
-export const RemotePollerConfigMapSchema = z.record(RemotePollerConfigSchema);
+export const RemotePollerConfigMapSchema = z.record(
+  RemotePollerConfigSchema.or(z.undefined())
```
Comment (rsioss) [inline src/lib/types/RemoteConnectionConfig.ts:49]: I wonder why the `.or(z.undefined())` was added here. did you do that intentionally or was it an automated eslint change?

**Review by rsioss — APPROVED** [review]
two small questions, but :shipit:!

File: src/lib/types/RemoteConnectionConfig.ts (hunk 2 above)
Comment (RossWilliams) [reply src/lib/types/RemoteConnectionConfig.ts:49]: I added this, not every key has a value.

Comment (rsioss) [reply src/lib/types/RemoteConnectionConfig.ts:49]: @RossWilliams every key _should_ have a value (that is the intent anyway). the `ConfigMapSchema` is the map of config id to config entry in stash:

```typescript
  "my-ftp-id": {
    // remote poller config details
  }
```

what would the use case be for not having a value to go along with this key?

**Pull request merged**